datastructures/clockchain.py

```diff
@@ -32,6 +37,81 @@ def __init__(self):
         genesis_dict = self.json_tick_to_chain_tick(tick)
         self.chain.put(genesis_dict)
 
+    def fork_pool_size(self):
+        return len(self.fork_pool)
+
+    def copy_chain(self, peer_addr):
+        logger.debug("Requesting chain from peer " + str(peer_addr))
+        peer_url = self.networker.reverse_peers.get(peer_addr, None)
+
+        if not peer_url:
+            logger.debug("Peer unknown, aborting chain request")
+            return False
+
+        logger.debug("Requesting chain from netloc " + str(peer_url))
+        r = requests.get(str(peer_url) + "/info/clockchain")
+
+        try:
+            alt_chain = r.json()['chain']
+        except KeyError:
+            logger.debug("Received invalid response from " + peer_addr + "; could not copy chain")
+            return False
+
+        alt_chain_is_valid = validate_clockchain(alt_chain)
+
+        if alt_chain_is_valid:
+            logger.debug("Received valid chain with " + str(len(alt_chain)) + " ticks")
+            logger.debug("Clearing current chain")
+            with self.chain.mutex:
+                self.chain.queue.clear()
+            logger.debug("Copying over new chain")
+            for tick in alt_chain:
+                logger.debug("Copying tick " + str(list(tick.keys())[0]))
+                self.chain.put(tick)
+            logger.debug("Chain copied")
+
+        logger.debug("Returning " + str(alt_chain_is_valid))
+        return alt_chain_is_valid
+
+    def resync(self):
+        alt_prev_ticks = list(set(tick['prev_tick'] for tick in self.fork_pool.values()))
+        logger.debug("Resyncing, alternative references found to ticks:")
+        for ref in alt_prev_ticks:
+            logger.debug(str(ref))
+
+        if len(alt_prev_ticks) > 1:
+            logger.debug("More than one alternative reference found, calculating majority")
+            ref_counts = [(prev_tick, alt_prev_ticks.count(prev_tick)) for prev_tick in alt_prev_ticks]
+            majority_prev_tick = sorted(ref_counts, key=lambda tup: tup[1], reverse=True)
```

Comment:
Again, can't I just fake massive amounts of references to a nonexistent tick or rogue tick in order to force your client to always accept my version of the truth? Or at the very least enable a DDOS attack that would leave you permanently unsynced